tests/app/main/test_suppliers.py

```diff
@@ -1982,3 +1982,63 @@ def test_one_open_one_closed_framework(self, data_api_client):
             # Check the right calls to action are there
             assert 'Create a supplier account' in data
             assert 'Get notifications when applications are opening' in data
+
+
+@mock.patch("app.main.views.suppliers.data_api_client", autospec=True)
+class TestSupplierEditOrganisationSize(BaseApplicationTest):
+    def test_edit_organisation_size_page_loads(self, data_api_client):
+        with self.app.test_client():
+            self.login()
+
+            res = self.client.get("/suppliers/organisation-size/edit")
+            assert res.status_code == 200, 'The edit organisation-size page has not loaded correctly.'
+
+    def test_no_selection_triggers_input_required_validation(self, data_api_client):
+        with self.app.test_client():
+            self.login()
+
+            res = self.client.post("/suppliers/organisation-size/edit")
+            doc = html.fromstring(res.get_data(as_text=True))
+            error = doc.xpath('//span[@id="error-organisation_size"]')
+
+            assert len(error) == 1, \
+                'Only one validation message should be shown.'
+
+            assert error[0].text.strip() == 'You must choose an organisation size.', \
+                'The validation message is not as anticipated.'
+
+    @pytest.mark.parametrize('size', (None, 'micro', 'small', 'medium', 'large'))
```

Comment:
We could test that an unexpected value posted here doesn't call the API, e.g. `ginormous` or something shouldn't be saved (I don't think this is enforced at the API level, so we will to enforce it at the frontend POST route here).

Reply:
I'm going to add an enum to the API schema to enforce this.  But still, would be nice to not make the call in the first place if someone posts something stupid.

Reply:
OK so the values are enforced by the API model, so API will error for a bad value, so we probably don't need to validate here after all.

Reply:
Yeah sorry, it is enforced by the API so I didn't bother. I could add a comment if you think that would prevent any confusion in the future?